src/Common/Extensions/LoggerAiExtensions.cs

```diff
@@ -43,6 +48,24 @@ public static void LogTrace(this ILogger logger, string message, IDictionary<str
             GetTelemetryClient().TrackTrace(message, props);
         }
 
+        /// <summary>
+        /// Logs a long message in chunks, with each chunk limited in length to 2^15.
+        /// </summary>
+        /// <param name="logger">logger instance</param>
+        /// <param name="level">log level. will apply to all chunks</param>
+        /// <param name="header">chunk header. will be followed by chunk index, a colon, and a line break</param>
+        /// <param name="message">long message to be chunkified and logged</param>
+        public static void LogLongMessage(this ILogger logger, LogLevel level, [NotNull] string header, string message)
+        {
+            int maxChunkLen = AiMessageLengthLimit - header.Length - 16; // 16 should cover for the header formatting
+            int i = 0;
```

Comment:
nit: since `chunks` is a `IList`, you can use indexes and avoid this

Reply:
true, but I think it's cleaner with `foreach` rather than `for`, WDYT?